unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -41,9 +98,21 @@ filtering_map groupDuplicates (const std::vector<std::string>& dataSource, Filte
     return grouped;
 }
 
+/**
+ * @todo Implement function that fill find duplicated files under the directory recursively
+ * @param rootPath - directory to check
+ * @return list of duplicated files, grouped by content
+ */
 std::vector<std::vector<std::string> > findDuplicates(const std::string &rootPath)
 {
-    // filter by size
-    // filter by content
-    // flatten
+    auto fileList = listFiles(rootPath);
+
+    auto grouped = groupDuplicates(fileList,
```

Comment:
also please group by size first as an optimization
see how it should look like:

```
    const std::vector<std::string>& files = listFiles(rootPath);
    auto get_size = [](const std::string& file) {
        std::error_code success;
        return fs::file_size(file, success);
    };

    auto sizeGroups = groupDuplicates(files, get_size);
    auto duplicatedSizes = flattenGrouped(sizeGroups);

    auto hashGroups = groupDuplicates(duplicatedSizes, xx_hash);
    std::vector<std::vector<std::string>> duplicates;
    for (const auto& sameHashGroup : hashGroups)
    {
        duplicates.push_back(sameHashGroup.second);
    }

    return duplicates;

```